src/components/app/App.js

```diff
@@ -0,0 +1,343 @@
+
+  const handleLoginClick = () => setIsLoginPopupOpen(true);
+
+  const closeAllPopups = () => {
+    setIsSignUpPopupOpen(false);
+    setIsLoginPopupOpen(false);
+    if (window.innerWidth < 520) {
+      noScroll();
+    }
+    history.push('/');
+  };
+
+  const toggleNoScroll = () => html.classList.toggle('no-scroll');
+
+  const noScroll = () => html.classList.add('no-scroll');
+
+  // ! Header handling
+  // * Handling the logout click
+  const handleLogout = () => {
+    setLoggedIn(false);
+    setIsHomePage(true);
+    setSavedArticles([]);
+    localStorage.removeItem('jwt');
+    history.push("/");
+  };
+
+  // * getting articles
+  React.useEffect(() => {
+    const closeByEscape = (evt) => {
+      if (evt.key === 'Escape') {
+        closeAllPopups();
+      }
+    };
+
+    document.addEventListener('keydown', closeByEscape);
+    return () => document.removeEventListener('keydown', closeByEscape);
+    // eslint-disable-next-line
+  }, []);
+
+  // ! Adding event listener for the page
+  // ! Mouse event
+  React.useEffect(() => {
+    const closeByClick = (evt) => {
+      if (evt.target.classList.contains("popup")) {
+        closeAllPopups();
+      }
+    }
+
+    document.addEventListener('mouseup', closeByClick);
+    return () => document.removeEventListener('mouseup', closeByClick);
+  });
+
+  // * Handling login form submit
+  const handleLoginSubmit = (email, password) => {
+    usersApiOBJ
+      .login({ email, password })
+      .then((data) => {
+        if (data.jwt) {
+          localStorage.setItem('jwt', data.jwt);
+        }
+        if (data.user._id) {
+          findUserInfo();
+        }
+      })
+      .catch((err) => {
+        console.log(`Error type: ${err.message}`);
+        setLoggedIn(false);
+      });
+  };
+
+  const findUserInfo = () => {
+    usersApiOBJ
+      .getCurrentUser(localStorage.getItem('jwt'))
+      .then((data) => {
+        if (data) {
+          setCurrentUser(data);
+          setLoggedIn(true);
+        } else {
+          setLoggedIn(false);
+        }
+      })
+      .catch((err) => {
+        if (err) {
+          console.log(`Error type: ${err.message}`);
+          setLoggedIn(false);
+        }
+      })
+      .finally(() => {
+        closeAllPopups();
+      });
+  }
+
+  // * Handling signup form submit
+  const handleSignupSubmit = (email, password, username) => {
+    usersApiOBJ
+      .signUp({ email, password, username })
+      .catch((err) => {
+        console.log(`Error type: ${err.message}`)
+      })
+      .finally(() => {
+        closeAllPopups();
+        setIsLoginPopupOpen(true);
+      })
+  };
+
+  // * checking if should auto-login
+  const isAutoLogin = () => {
+    const jwt = localStorage.getItem('jwt');
+    if (jwt) {
+      auth.checkToken(jwt)
+        .then((user) => {
+          if(user){
+            setCurrentUser(user);
+            setLoggedIn(true);
+          }
+        });
```

Comment:
You need to `catch` possible errors at the very end of any server request.

**Please, correct it everywhere**